control-plane/subcommand/server-acl-init/connect_inject.go

```diff
@@ -96,16 +96,13 @@ func (c *Command) createAuthMethodTmpl(authMethodName string, useNS bool) (api.A
 
 	var saSecret *apiv1.Secret
 	var secretNames []string
-	if len(authMethodServiceAccount.Secrets) == 0 {
-		// In Kube 1.24+ there is no automatically generated long term JWT token for a ServiceAccount.
-		// Furthermore, there is no reference to a Secret in the ServiceAccount. Instead we have deployed
-		// a Secret in Helm which references the ServiceAccount and contains a permanent JWT token.
-		secretNames = append(secretNames, c.withPrefix("auth-method"))
-	} else {
-		// ServiceAccounts always have a SecretRef in Kubernetes < 1.24. The Secret contains the JWT token.
-		for _, secretRef := range authMethodServiceAccount.Secrets {
-			secretNames = append(secretNames, secretRef.Name)
-		}
+	// In Kube 1.24+ there is no automatically generated long term JWT token for a ServiceAccount.
+	// Furthermore, there is no reference to a Secret in the ServiceAccount. Instead we have deployed
+	// a Secret in Helm which references the ServiceAccount and contains a permanent JWT token.
+	secretNames = append(secretNames, c.withPrefix("auth-method"))
```

Comment:
I'm wondering about the order here for the Openshift case where it injects secrets into the ServiceAccount.
Are those supposed to hold the JWT token that we're supposed to use? if so I think we just need to re-order these four lines so that we attempt to read the Helm deployed Secret last?